core/src/apps/ethereum/yielding.py

```
@@ -110,19 +128,25 @@ def _prepare_vault_tx(
+            not isinstance(amount, int)
             or not isinstance(receiver_bytes, bytes)
             or (owner_bytes is not None and not isinstance(owner_bytes, bytes))
             or int.from_bytes(msg.value, "big") != 0
             or amount == 0
         ):
             raise ValueError
     except (ValueError, EOFError, InvalidFunctionCall):
-        raise DataError("Invalid data for ERC-4626 vault transaction")
+        raise DataError(
+            f"Invalid data for ERC-4626 vault transaction. Remaining data count: {data_reader.remaining_count()}, amount: {amount}, eth_Value: {int.from_bytes(msg.value, 'big')}"
+        )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)